src/views/Teams/TeamDetails.js

```diff
@@ -0,0 +1,30 @@
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import TeamDetailsCard from '../../components/Teams/TeamDetailsCard';
+import { fetchTeamById } from '../../services/teams';
+
+export default function TeamDetails() {
+  const params = useParams();
+  const teamId = params.id;
+  const [team, setTeam] = useState({});
+  const [teamPlayers, setTeamPlayers] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const data = await fetchTeamById(teamId);
+      setTeam(data);
+      setTeamPlayers(data.players);
+      setLoading(false);
+    };
+    fetchData();
+  }, [teamId]);
+
+  if (loading) return (<div className="loader"></div>);
```

Comment:
Excellent use of loading state to prevent the race condition of our data mounting versus displaying! Nice job!